server/service/src/sync/translations/special/name_merge.rs

```diff
@@ -56,7 +60,57 @@ impl SyncTranslation for NameMergeTranslation {
             })
             .collect();
 
-        Ok(Some(IntegrationRecords::from_upserts(upsert_records)))
+        let name_store_join_repo = NameStoreJoinRepository::new(connection);
+        let name_store_joins_for_delete = name_store_join_repo.query(Some(
+            NameStoreJoinFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_delete)),
+        ))?;
+        let name_store_joins_for_keep = name_store_join_repo.query(Some(
+            NameStoreJoinFilter::new().name_id(EqualFilter::equal_to(&data.merge_id_to_keep)),
```

Comment:
Nit: there is also `query_by_filter` if you want to avoid wrapping it into an Option

Reply:
shot 76278e8e4233